source/arm9/hw_math.c

```diff
@@ -0,0 +1,35 @@
+
+
+#include <math.h>
+
+
+
+f32 hw_sqrtf(f32 x){
+    union{f32 f; u32 i;}xu;
+    xu.f=x;
+    u64 mantissa=xu.i & ((1<<23)-1);
+    //check if exponent is odd
+    //before subtracting 127 exponent was even if it is odd now
+    //therefore check if last digit is 0 
+    mantissa=(mantissa+(1<<23))<<   ( (( xu.i & (1<<23))==0  )      +25 );
+    REG_SQRTCNT = SQRT_64;
+    REG_SQRT_PARAM = mantissa;
+    u32 raw_exponent= (xu.i & (0xff<<23));
+    if (raw_exponent > 0 ) {
```

Comment:
In general, could you try to imitate the code style of the rest of the library? For example, don't leave that space after the 0, don't leave that curly bracket in that line (move it to the next one). Leave one space after `//` in your comments, always leave one space on both sides of a `=` or other operators.